src/js/plugins/plugin.text_selection.js

```diff
@@ -0,0 +1,204 @@
+//@ts-check
+import { isFirefox } from './tts/utils';
+
+const BookReader = /** @type {typeof import('../BookReader').default} */(window.BookReader);
+
+class TextSelectionPlugin {
+
+  constructor(avoidTspans = isFirefox()) {
+    /**@type {PromiseLike<JQuery<HTMLElement>>} */
+    this.djvuPagesPromise = null;
+    // Using text elements insted of tspans for words because Firefox does not allow svg tspan strech.
+    // Tspans are necessary on Chrome because they prevent newline character after every word when coping
+    this.svgParagraphElement = "text";
+    this.svgWordElement = "tspan";
+    this.onFirefox = avoidTspans
+    if(avoidTspans) {
+      this.svgParagraphElement = "g";
+      this.svgWordElement = "text";
+    }
+  }
+
+  /**
+   * @param {string} ocaid
+   */
+  init(ocaid) {
+    this.djvuPagesPromise = $.ajax({
+      type: "GET",
+      url: `https://cors.archive.org/cors/${ocaid}/${ocaid}_djvu.xml`,
+      dataType: "xml",
+
+      error: function (e) {
+        return undefined;
+      }
+    }).then(function (xmlMap) {
+      if (xmlMap != undefined) {
+        return $(xmlMap).find("OBJECT");
+      }
+    });
+  }
+
+  /**
+   * @param {number} index
+   * @returns {Promise<HTMLElement>}
+   */
+  async getPageText(index) {
+    return (await this.djvuPagesPromise)[index];
+  }
+
+  /**
+   * Intercept copied text
+   * @param {JQuery} $container
+   */
+  interceptCopy($container) {
+    $container.each((i, pageContainer) => {
+      pageContainer.addEventListener('copy', (event) => {
+        const selection = document.getSelection();
+        event.clipboardData.setData('text/plain', selection.toString());
+        event.preventDefault();
+      });
+    });
+  }
+
+  /**
+   * Deselect all text after clicking elswhere in the BRcontainer
+   * @param {JQuery} $container
+   */
+  deselectOnClick($container) {
```

Comment:
This isn't working; clicking anywhere after selecting causes a page flip

Reply:
That function is only called for the 1page mode for now

Reply:
Ah; Can we add it 2up as well? Or is it more complicated than that?